### PR TITLE
Backend frontend connection

### LINKED ISSUE #96: feature/link-backend-to-frontend
## User Story: (10 points)
As any user, I want to be able to maintain my changes permanently between visits.

## Brief Description
Connected Google Cloud backend to React app frontend.

## Feature Requirements
- Use Axios for HTTP requests (GET, POST, DELETE)
- Maintain products, users, sellers, orders, and brands in a database
- Ensure same functionality as test data
- Replace uses of test data with correct backend API requests

### PR DESCRIPTION
**Pull request by Adamo-O**
## Brief Description
Linked backend Google Cloud with frontend React app.

## Linked Issues
- closes #96 

### Frontend
- Used Axios for HTTP Requests (GET, POST, DELETE)
- Converted use of test data to appropriate API request
- Features using Google Cloud backend:
   - View products in grid
   - View product details in page
   - Sellers can view products
   - Sellers can modify existing products
   - Sellers can add products
   - Sellers can delete products

### Backend
- Added a route that lists all sellers
- Brand descriptions are not required in requests
- Brand names must be unique
- Brands deleted if they have no attached products

## Additional context
For a guide on all requests and routes to the backend, visit the [How to use the backend guide](https://github.com/SOEN-341-Project/Group-P-soen341project2022/wiki/How-to-use-the-backend).

### CODE REVIEW

**Review by aspyridakos — DISMISSED** [review]
Nice! ⚡ 💯 ⚡

**Review by emma-amar — APPROVED** [review]
looks great!! thanks for setting it all up !!! :-))

**Review by oumbarry — APPROVED** [review]
Great work

**Review by jungsoolee1 — DISMISSED** [review]
Nicely explained. Thank you so much!

**Review by aspyridakos — APPROVED** [review]

**Review by jungsoolee1 — APPROVED** [review]
Good job

File: online-shopping-website/frontend/src/Components/Products/SideNav.js [hunk 1]
```diff
@@ -124,14 +124,14 @@ function BrandsCheckbox(props) {
         props.onCheckboxChange('Brand', event.target.name, event.target.checked);
     };
 
-    function iterateSellers() {
```
Comment (aspyridakos) [inline online-shopping-website/frontend/src/Components/Products/SideNav.js:127]: Should be called iterateBrands

File: online-shopping-website/frontend/src/Components/Products/SideNav.js [hunk 2]
```diff
@@ -124,14 +124,14 @@ function BrandsCheckbox(props) {
...
-        return TestData.brands.map((brand, index) => {
+    function iterateSellers(props) {
```
Comment (aspyridakos) [inline online-shopping-website/frontend/src/Components/Products/SideNav.js:127]: same as above

**Review by aspyridakos — CHANGES_REQUESTED** [review]

**Review by oumbarry — APPROVED** [review]
Good pr!

**Review by aspyridakos — DISMISSED** [review]

File: online-shopping-website/frontend/src/Components/Products/SideNav.js (hunk 2 above)
Comment (KooshaG) [reply online-shopping-website/frontend/src/Components/Products/SideNav.js]: It's changed!

File: online-shopping-website/frontend/src/Components/Products/SideNav.js (hunk 1 above)
Comment (KooshaG) [reply online-shopping-website/frontend/src/Components/Products/SideNav.js:127]: It's changed!

**Review by aspyridakos — APPROVED** [review]
Looking good:)

**Pull request merged**